TypeGap/GapApiGenerator.cs

```diff
@@ -416,7 +416,11 @@ protected virtual ApiParamDesc[] ValidateParameters(List<ApiParamDesc> parameter
 
             postParam = post;
             modelParam = null;
-            return parameters.Except(new[] { post }).ToArray();
+
+            return (post == null ?
+                    parameters :
+                    parameters.Except(new[] { Nullable.GetUnderlyingType(post.ParameterType) != null ? null : post }))
+                .ToArray();
```

Comment:
This is not right. First of all, it ends up returning the same parameter twice - once through `postParam` and once again through the returned array.

The real problem is that `Nullable<long>` is considered as a `postPossibilities`, which it shouldn't be. You should revert this and fix `TypeConverter.IsComplexType`.

Reply:
True, really nice comment. Fixed by adding logic to the complextype